tools/check_codex_compat.py

```
@@ -0,0 +1,232 @@
+    return result.stdout.strip()
+
+
+def _assert_plugin(codex: Path, expected_version: str) -> None:
+    result = _capture([str(codex), "plugin", "list", "--json"])
+    if result.returncode != 0:
+        detail = (result.stderr or result.stdout).strip()
+        raise CompatError(f"codex plugin list failed ({result.returncode}): {detail}")
+    try:
+        payload = json.loads(result.stdout)
+    except json.JSONDecodeError as exc:
+        raise CompatError(f"codex plugin list did not return JSON: {exc}") from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)